bitcoin/src/blockdata/script.rs

```diff
@@ -139,13 +139,19 @@ pub struct Script([u8]);
 ///
 /// [deref coercions]: https://doc.rust-lang.org/std/ops/trait.Deref.html#more-on-deref-coercion
 #[derive(Default, Debug, Clone, PartialOrd, Ord, PartialEq, Eq, Hash)]
-pub struct ScriptBuf(Vec<u8>);
+pub struct ScriptBuf<T = Generic>(PhantomData<T>, Vec<u8>);
+
+/// Marker for a generic [`ScriptBuf`].
+pub type Generic = ();
+
+/// Marker for a [`ScriptBuf`] that is used as script code.
+pub type ScriptCode = ();
```

Comment:
We have to use empty enums, not type alias. The type alias allows you to mix them up and the point of script tagging is to avoid mixing up things.

Reply:
Face palm - this is my first go at this sort of "tagging", as you have probably guessed :)